sql/api/src/main/scala/org/apache/spark/sql/types/DataType.scala

```diff
@@ -72,6 +72,64 @@ abstract class DataType extends AbstractDataType {
   /** The pretty (i.e. indented) JSON representation of this data type. */
   def prettyJson: String = pretty(render(jsonValue))
 
+  /** Util to recursively form JSON representation of data type, used for DESCRIBE AS JSON.
+   * Differs from `json` by overriding the JSON representation of some types. */
+  def jsonType: String = {
+    this match {
+      case arrayType: ArrayType =>
+        val elementTypeJson = arrayType.elementType.jsonType
+        s"""{"type": "array","elementType": $elementTypeJson,
+           |"containsNull": ${arrayType.containsNull}}""".stripMargin
+
+      case mapType: MapType =>
+        val keyTypeJson = mapType.keyType.jsonType
+        val valueTypeJson = mapType.valueType.jsonType
+        s"""{"type": "map","keyType": $keyTypeJson,"valueType": $valueTypeJson,
+           |"valueContainsNull": ${mapType.valueContainsNull}}""".stripMargin
+
+      case structType: StructType =>
+        val fieldsJson = structType.fields.map { field =>
+          val fieldTypeJson = field.dataType.jsonType
+          s"""{"name": "${field.name}", "type": $fieldTypeJson, "nullable": ${field.nullable}}"""
+        }.mkString(", ")
+        s"""{"type": "struct", "fields": [$fieldsJson]}"""
+
+      case decimalType: DecimalType =>
+        s"""{"type": "decimal", "precision": ${decimalType.precision},
+           |"scale": ${decimalType.scale}}""".stripMargin
+
+      case varcharType: VarcharType =>
+        s"""{"type": "varchar", "length": ${varcharType.length}}"""
+
+      case charType: CharType =>
+        s"""{"type": "char", "length": ${charType.length}}"""
+
+      // Only override TimestampType; TimestampType_NTZ type is already timestamp_ntz
+      case _: TimestampType =>
+        s"""{"type": "timestamp_ltz"}"""
```

Comment:
interpolation is not needed here:
```suggestion
        """{"type": "timestamp_ltz"}"""
```